llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -435,6 +433,98 @@ def IREELinalgExt_TopkOp : IREELinalgExt_Op<"topk",[
   }];
 }
 
+def IREELinalgExt_PackOp : IREELinalgExt_Op<"pack", [
+  DeclareOpInterfaceMethods<ReifyRankedShapedTypeOpInterface>,
+  DeclareOpInterfaceMethods<LinalgExtInterface>,
+  DeclareOpInterfaceMethods<TilingInterface,
+    ["getIterationDomain",
+     "getLoopIteratorTypes",
+     "generateScalarImplementation"]>,
+  DeclareOpInterfaceMethods<MemoryEffectsOpInterface>
+]>{
+  let summary = "pack operation"; 
+  let description = [{
+    The pack operation takes a shaped type as input, tiles, and transposes it
+    based on `dims_pos` and `inner_tiles`. The former carries the indices of the
+    dimension to tile, while the latter the tiling factor. A tile factor can
+    either be a constant or an index type. We handle only full tiles now; it is UB
+    if the tile does not perfectly divide the dimension. We will relax this
+    constraint in the future when we add pad semantics to the operation.
+  }];
+
+  let arguments = (ins Variadic<AnyShaped>:$inputs,
```

Comment:
Is there a reason to have `Variadic<AnyShaped>` instead of just a single `input`?

Reply:
You are right, it should be a single input and single output operation. I went for the variadic because `IREELinalgExt_Op` implements the `AttrSizedOperandSegments`. Other ops like `ReverseOp` do the same. I can definitely pull out `AttrSizedOperandSegments` from `IREELinalgExt_Op` and have a single input/output operand. I would like to do that in a follow-up PR.